addons/sourcemod/scripting/mapchooser_extended.sp

```diff
@@ -870,39 +867,38 @@ public void Event_WeaponRank(Handle event, const char[] name, bool dontBroadcast
 	}
 }
 
-/* You ask, why don't you just use team_score event? And I answer... Because CSS doesn't. */
-public Action Event_PreRoundEnd(Handle event, const char[] name, bool dontBroadcast)
+public void Event_WinPanel(Handle event, const char[] name, bool dontBroadcast)
 {
-	if (!g_Cvar_EndOfMapInfo)
-		return Plugin_Handled;
-
-	int timeLeft;
-	if(!GetMapTimeLeft(timeLeft))
-		return Plugin_Handled;
-
-	if(timeLeft >= 0)
-		return Plugin_Handled;
+	if (g_Cvar_EndOfMapInfo.IntValue != 1)
+		return;
 
 	char nextMap[64];
 	if(!GetNextMap(nextMap, sizeof(nextMap)))
-		return Plugin_Handled;
+		return;
+
+	Handle g_hHud = CreateHudSynchronizer();
 
 	if (g_hHud != INVALID_HANDLE)
 	{
-		SetHudTextParams(-1.0, 0.01, 4.0, 255, 71, 1, 1, 1, 4.0, 0.6, 0.6);
+		SetHudTextParams(-1.0, 0.89, 7.0, 255, 71, 1, 1, 1, 4.0, 0.6, 0.6);
 		for(int i = 1; i <= MaxClients; i++)
 		{
-			if(!IsClientInGame(i) || IsFakeClient(i) || IsClientObserver(i))
+			if(!IsClientInGame(i) || IsFakeClient(i))
 				continue;
 
 			ClearSyncHud(i, g_hHud);
 			ShowSyncHudText(i, g_hHud, "Next Map: %s", nextMap);
 		}
 	}
+	
+	CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);	
+	CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);	
+	CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);
 
-	return Plugin_Handled;
+	delete g_hHud;
```

Comment:
No because using `delete` in void function call `return;`
Exemple :: Bad version
```cpp
CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);    
delete g_hHud; // The next part of the code will not be executed if it in void function due to delete who calling return;
CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);    
CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);
```
Work fine version because delete call return;
```cpp
CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);    
CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);    
CPrintToChatAll("{lightgreen}Next Map: {green}%s", nextMap);
delete g_hHud;
```

Reply:
i will link the commit i dont think you see what i meant

Reply:
https://github.com/srcdslab/sm-plugin-mapchooser_extended/commit/323d10a2dfedbdc6bde9ed7b6ba832f3fffc2994